scripts/override-start.js

```diff
@@ -24,8 +24,26 @@ const overrideWebpackConfig = ( config, devServer ) => {
 	// Replace the react-dev-utils webpackHotDevClient with a version patched to
 	// correctly detect the dev server host & port for socket requests.
 	const hotClient = require.resolve( 'react-dev-utils/webpackHotDevClient' );
-	const hotClientIndex = config.entry.indexOf( hotClient );
-	config.entry.splice(hotClientIndex, 1, require.resolve( '../overrides/webpackHotDevClient' ) );
+
+	if ( Array.isArray( config.entry ) ) {
+		const hotClientIndex = config.entry.indexOf( hotClient );
+		if ( hotClientIndex >= 0 ) {
+			config.entry.splice( hotClientIndex, 1, require.resolve( '../overrides/webpackHotDevClient' ) );
+		}
+	} else if ( config.entry instanceof Object ) {
+		Object.entries( config.entry ).forEach( ( key, entry ) => {
+			if ( entry === hotClient ) {
+				config.entry[key] = require.resolve( '../overrides/webpackHotDevClient' );
+			}
+
+			if ( Array.isArray( entry ) ) {
```

Comment:
`else if`

Reply:
I don't think it strictly needs to be an `else if`, it won't affect the behaviour there. I usually find this style a bit more readable